plugins/google_analytics_reports_plugin_query_google_analytics.inc

```diff
@@ -217,42 +227,50 @@ class google_analytics_plugin_query_google_analytics extends views_plugin_query
    * Values to set: $view->result, $view->total_rows, $view->execute_time,
    * $view->current_page.
    */
-  function execute(&$view) {  
+  function execute(&$view) {
+    // Initial check to see if we should attempt to run the query.
+    if (!variable_get('google_analytics_reports_api_access_token', FALSE)) {
+      drupal_set_message(t('You must <a href="@url">authorize</a> Drupal to use your Google Analytics account before you can view reports.',
+        array('@url' => url('admin/config/system/google-analytics-reports-api'))), 'warning', FALSE);
+      return;
+    }
+
     $query = $view->build_info['query'];
     $count_query = $view->build_info['count_query'];
     $start = microtime(TRUE);
 
-    /* Query for total number of items */
+    // Query for total number of items.
     $count_query['max_results'] = 9999;
     $count_query['start_index'] = 1;
-    $count_feed = google_analytics_api_report_data($count_query);
+    $count_feed = google_analytics_reports_api_report_data($count_query);
     if (!is_null($count_feed->error)) {
       return;
     }
     $this->pager->total_items = count($count_feed->results->rows);
     $this->pager->update_page_info();
 
-    /* Adjust based on the pager's modifications to limit and offset */
+    // Adjust based on the pager's modifications to limit and offset.
     if (!empty($this->limit) || !empty($this->offset)) {
       $query['max_results'] = intval(!empty($this->limit) ? $this->limit : 1000);
       $query['start_index'] = intval(!empty($this->offset) ? $this->offset : 0) + 1;
     }
 
-    $feed = google_analytics_api_report_data($query);
-    
+    $feed = google_analytics_reports_api_report_data($query);
+
     $views_result = array();
```

Comment:
Could probably remove this (one of our old additions, I believe), since it's now checked below... Shrug?